dominion/src/main.rs

```diff
@@ -56,9 +56,25 @@ fn big_money(arena: &mut Arena, player_id: usize) {
 
     play_all_treasures(arena, player_id);
 
-    let _ = arena.select_card(player_id, CardKind::Province, Location::Supply);
-    let _ = arena.select_card(player_id, CardKind::Gold, Location::Supply);
-    let _ = arena.select_card(player_id, CardKind::Silver, Location::Supply);
+    arena
+        .buy_card(player_id, CardKind::Province)
+        .map_err(|_| arena.buy_card(player_id, CardKind::Gold))
+        .map_err(|_| arena.buy_card(player_id, CardKind::Silver))
+        .or_else(|_| -> Result<(), ()> {
+            match arena.turn_phase() {
+                TurnPhase::Action { .. } => {
+                    panic!("Expected TurnPhase::Buy but got TurnPhase::Action.")
+                }
+                TurnPhase::Buy {
+                    remaining_buys,
+                    total_wealth,
+                } => {
+                    assert!(total_wealth < CardKind::Silver.cost());
+                    Ok(())
+                }
+            }
+        })
+        .unwrap();
 
     arena.end_buy_phase(player_id).unwrap();
```

Comment:
Since we keep track of the current state (player, turn) and all the other methods check if the game is in the right state, this function can simply be `end_phase()` which instead of erroring just does the right thing?